src/aics-image-viewer/components/App/index.tsx

```diff
@@ -179,6 +179,8 @@ const App: React.FC<AppProps> = (props) => {
 
   // State for image loading/reloading
 
+  // `true` when this is the initial load of an image
+  const initialLoadRef = useRef(true);
```

Comment:
is this a standard way to keep a one-shot variable?  Why a ref?  is it sort of like state that doesn't force a re-render?

Reply:
Refs are different from state in two ways: they don't force a re-render, and they are always up-to-date within closures. `onChannelDataLoaded` is called from within a closure that is created alongside `image` and never changed thereafter, so a ref is the right tool for the job.